### DISCUSSION TITLE
Python 3.12 new features

### DISCUSSION THREAD

**Discussion by AlexanderPodorov** [pydantic/pydantic#6704] [Question] [answered]
Hello, thanks a lot for the great library!
Just wanted to make sure, is support for the PEP 692 and PEP 695 of Python 3.12 planned sometime soon? I mean once 3.12 is released, can we use its new features right away? Thanks.

**Comment by samuelcolvin** [comment]
No idea if they work out of the box, I guess not.

But I believe pydantic works on 3.12, so please try them and let us know how you get on.

**Comment by AlexanderPodorov** [comment]
Yes, I’m not sure about PEP 692, but PEP 695 definitely won’t.
Just wanted to know what are the plans about that. Thanks!

**Comment by samuelcolvin** [comment, marked as answer]
We definitely plan to support them, not sure how quickly.

We also plan to support the new turbofish syntax for generics (can't remember the pep) which will go into 3.13, when that comes out.

**Comment by AlexanderPodorov** [comment]
I think you meant [PEP 695](https://peps.python.org/pep-0695) which will be included into 3.12 and [PEP 696](https://peps.python.org/pep-0696) which will come with 3.13 (supposedly)
